EKrut-Server/src/ekrut/server/managers/ServerShipmentManager.java

```diff
@@ -1,53 +1,148 @@
-		
-		// estimate delivery time.
+		Order order = orderDAO.fetchOrderById(orderId);
+
+		// In case the order null an exception will be thrown
+		if (order == null)
+			return new ShipmentResponse(ResultType.NOT_FOUND);
+
+		if (order.getType() != OrderType.SHIPMENT)
+			return new ShipmentResponse(ResultType.INVALID_INPUT);
+
+		if (order.getStatus() != OrderStatus.AWAITING_DELIVERY)
+			return new ShipmentResponse(ResultType.INVALID_INPUT);
+
+		// Estimate delivery time.
 		LocalDateTime estimateDeliveryTime = ShipmentManagerUtils.estimatedArrivalTime(date, clientAddress);
-	
-		// create shipping in DB.
-		//ShipmentDAO. // TBD Q.Nir
-		
-		return null;
-	}
-	
-	
+		order.setDueDate(estimateDeliveryTime);
 
-	
-	public boolean confirmDelivery() {
-		return false;
-		
-	}
-	
+		// Q.Nir , do we need to add another method in case the worker cancel?
+		// and for this we will need to add CANCELED_SHIPMENT in in OrderStatus
+		// what's happen in case the worker cancel the shipment? we can choose to cancel
+		// and continue with
+		// regular order process or we will delete the whole method
+		if (!orderDAO.updateOrderStatus(orderId, OrderStatus.AWAITING_DELIVERY))
+			return new ShipmentResponse(ResultType.UNKNOWN_ERROR);
 
-	public boolean confirmShipment() {
-		return false;
-		
+		return new ShipmentResponse(ResultType.OK);
 	}
-	
 
-	public boolean setDone() {
-		return false;
+	/**
+	 * Confirms the delivery of an order.
+	 * 
+	 * @param shipmentRequest the {@code ShipmentRequest} object containing the request details
+	 * @return a {@code ShipmentResponse} object with the result of the operation
+	 */
+	public ShipmentResponse confirmDelivery(ShipmentRequest shipmentRequest) {
+		int orderId = shipmentRequest.getOrderId();
+		Order order = orderDAO.fetchOrderById(orderId);
+		if (order == null)
+			return new ShipmentResponse(ResultType.NOT_FOUND);
+
+		if (order.getType() != OrderType.SHIPMENT)
+			return new ShipmentResponse(ResultType.UNKNOWN_ERROR);
+
+		if (order.getStatus() != OrderStatus.AWAITING_DELIVERY)
+			return new ShipmentResponse(ResultType.UNKNOWN_ERROR);
+
+		if (!orderDAO.updateOrderStatus(orderId, OrderStatus.DELIVERY_CONFIRMED))
+			return new ShipmentResponse(ResultType.UNKNOWN_ERROR);
+
+		return new ShipmentResponse(ResultType.OK);
 	}
-}
 
-	
+	/**
+	 * Marks an order as done.
+	 * 
+	 * @param shipmentRequest the {@code ShipmentRequest} object containing the
+	 *                        request details
+	 * @return a {@code ShipmentResponse} object with the result of the operation
+	 */
+	public ShipmentResponse setDone(ShipmentRequest shipmentRequest) {
+		int orderId = shipmentRequest.getOrderId();
+		Order order = orderDAO.fetchOrderById(orderId);
+		if (order == null)
+			return new ShipmentResponse(ResultType.NOT_FOUND);
+
+		if (order.getType() != OrderType.SHIPMENT)
+			return new ShipmentResponse(ResultType.UNKNOWN_ERROR);
 
+		if (order.getStatus() != OrderStatus.AWAITING_DELIVERY)
```

Comment:
This should be `DELIVERY_CONFIRMED` and the update below should change it to `DONE`.

Reply:
Fixed